api/src/main/java/org/pentaho/platform/api/genericfile/model/IGenericFileMetadata.java

```diff
@@ -19,5 +19,25 @@
  * {@code IGenericFile}'s metadata.
  */
 public interface IGenericFileMetadata {
+  /**
+   * Gets the metadata map.
+   *
+   * @return the metadata map
+   */
   Map<String, String> getMetadata();
```

Comment:
I believe we should add the `@NonNull` annotation to the getter result and setter argument.